src/border_agent/border_agent.cpp

```diff
@@ -210,6 +215,68 @@ void BorderAgent::Stop(void)
     UnpublishMeshCopService();
 }
 
+#if OTBR_ENABLE_EPSKC
+void BorderAgent::HandleEpskcStateChanged(void *aContext)
+{
+    BorderAgent *borderAgent = static_cast<BorderAgent *>(aContext);
+    if (otBorderAgentIsEphemeralKeyActive(borderAgent->mNcp.GetInstance()))
+    {
+        borderAgent->PublishEpskcService();
+    }
+    else
+    {
+        borderAgent->UnpublishEpskcService();
+    }
+}
+
+void BorderAgent::PublishEpskcService()
+{
+    otInstance              *instance = mNcp.GetInstance();
+    int                      port     = otBorderAgentGetUdpPort(instance);
```

Comment:
In spec the section `DNS-SD Service MeshCoP-ePSKc`

 ```The service MUST reside at a different UDP port number than the regular MeshCoP (_meshcop._udp) service.```

Reply:
Yes, `otBorderAgentGetUdpPort()` returns the port of `_meshcop-e._udp` service when the device is in ePSKc mode. See https://github.com/openthread/openthread/blob/4737231d64d262eb850e0da729d4b2df130033e1/include/openthread/border_agent.h#L198

@abtink But I think it's better to have a new API to return the ePSKc port independently. Otherwise, if there is a update to the `_mehscop._udp` service during the ePSKc mode, the wrong port will be used there. Thoughts?

Reply:
Border Agent may be using PSK (traditional commissioning model) or set up to use e-PSK, but cannot do not both at the same time. So it can accept connections on one port (at a time) depending on its mode. This port can be retrived using `otBorderAgentGetUdpPort()`.

Reply:
Thanks for the clarification,  the concern of using the same UDP ports would not happen in current behavior.